### PR TITLE
Initial API unit test case - Root endpoint

### PR DESCRIPTION
**Pull request by JenySadadia**
Fixes https://github.com/kernelci/kernelci-api/issues/36

api/test_api.py : Add test for root endpoint using fastapi TestClient
api/__init__.py : Add file to make api a package to import app
Error fix : importError: attempted relative import with no known parent package
docker/api/requirements-dev.txt : Add pytest and fastapi

Signed-off-by: Jeny Sadadia <jeny.sadadia@gmail.com>

### CODE REVIEW

**Comment by mgalka** [comment]
@JenySadadia Does this work for you?
I installed packages from requirements.txt and requirements-dev.txt
I tried running it with
```
pytest api
```
and I'm getting errors

```
========================================================================================= test session starts =========================================================================================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/mgalka/projects/kernelci/kernelci-api
collected 0 items / 1 error                                                                                                                                                                           

=============================================================================================== ERRORS ================================================================================================
__________________________________________________________________________________ ERROR collecting api/test_api.py ___________________________________________________________________________________
api/test_api.py:7: in <module>
    from .main import app
api/main.py:16: in <module>
    auth = Authentication(db)
api/auth.py:35: in __init__
    self._settings = Settings()
pydantic/env_settings.py:36: in pydantic.env_settings.BaseSettings.__init__
    ???
pydantic/main.py:406: in pydantic.main.BaseModel.__init__
    ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for Settings
E   secret_key
E     field required (type=value_error.missing)
```

**Comment by JenySadadia** [comment]
Hi @mgalka ,
You need to export .env file with SECRET_KEY.
This would work after that.

File: api/test_api.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+def test_root_endpoint():
+    """
+    Test Case : Test KernelCI API root endpoint
+    Expected Result :
+        HTTP Response Code 200 OK
+        JSON with 'message' key
+    """
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.json() == {"message": "KernelCI API"}
```
Comment (mgalka) [inline api/test_api.py:22]: We will probably need an instance of the `TestClient` in a few places so it maybe worth creating a fixture out of it.

```suggestion
@pytest.fixture
def client():
    return TestClient(app)


def test_root_endpoint(client):
    """
    Test Case : Test KernelCI API root endpoint
    Expected Result :
        HTTP Response Code 200 OK
        JSON with 'message' key
    """
    response = client.get("/")
    assert response.status_code == 200
    assert response.json() == {"message": "KernelCI API"}
```

**Review by mgalka — CHANGES_REQUESTED** [review]
A few minor things in the code. 

Apart from that it'll be good to add a section in README on how to run tests.

File: docker/api/requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+pytest==6.2.5
```
[the remark this answers is not in the document]
Comment (JenySadadia) [reply docker/api/requirements-dev.txt:1]: How about we use a shell script to install all the packages?
I have committed file ``` installPackages.sh ```
Please review.

[the remark this answers is not in the document]
Comment (JenySadadia) [reply docker/api/requirements-dev.txt:1]: We can have one shell script to setup prod env and another one for dev env.

**Review by mgalka — APPROVED** [review]
Looks good to me

**Pull request merged**